src/analysis/special_functions/log.lean

```diff
@@ -172,6 +172,32 @@ begin
   { rintro (rfl|rfl|rfl); simp only [log_one, log_zero, log_neg_eq_log], }
 end
 
+lemma log_le_sub_one_of_nonneg {x : ℝ} (hx : 0 < x) : log x ≤ x - 1 :=
+begin
+  have h := add_one_le_exp (log x),
+  rw exp_log hx at h,
+  rw le_sub_iff_add_le,
+  exact h,
+end
+
+lemma log_div_self_decreasing {x y : ℝ} (hex : exp 1 ≤ x) (hxy : x ≤ y) :
+  log y / y ≤ log x / x :=
+begin
+  have x_pos : 0 < x := (exp_pos 1).trans_le hex,
+  have y_pos : 0 < y := x_pos.trans_le hxy,
+  rw [div_le_iff y_pos, ←sub_le_sub_iff_right (log x), ←log_div (y_pos.ne') (x_pos.ne'),
+    ←mul_div_right_comm, mul_div_assoc],
+  have : log x * (y / x) - log x = log x * (y / x - 1), ring,
```

Comment:
I think that a `calc` proof will be more readable (though it may be a bit longer).